tests/integration/hugectr/test_nvt_hugectr_training.py

```diff
@@ -0,0 +1,380 @@
+            top_names=["relu1"],
+        )
+    )
+    model.add(
+        hugectr.DenseLayer(
+            layer_type=hugectr.Layer_t.InnerProduct,
+            bottom_names=["relu1"],
+            top_names=["fc2"],
+            num_output=128,
+        )
+    )
+    model.add(
+        hugectr.DenseLayer(
+            layer_type=hugectr.Layer_t.ReLU,
+            bottom_names=["fc2"],
+            top_names=["relu2"],
+        )
+    )
+    model.add(
+        hugectr.DenseLayer(
+            layer_type=hugectr.Layer_t.InnerProduct,
+            bottom_names=["relu2"],
+            top_names=["fc3"],
+            num_output=1,
+        )
+    )
+    model.add(
+        hugectr.DenseLayer(
+            layer_type=hugectr.Layer_t.BinaryCrossEntropyLoss,
+            bottom_names=["fc3", "label"],
+            top_names=["loss"],
+        )
+    )
+    model.compile()
+    model.summary()
+    model.fit()
+
+
+def _write_model_json(slot_sizes, total_cardinality):
+
+    config = json.dumps(
+        {
+            "inference": {
+                "max_batchsize": 64,
+                "hit_rate_threshold": 0.6,
+                "dense_model_file": MODEL_DIR + "test_model/1/_dense_1900.model",
+                "sparse_model_file": MODEL_DIR + "test_model/1/0_sparse_1900.model",
+                "label": 1,
+                "input_key_type": "I64",
+            },
+            "layers": [
+                {
+                    "name": "data",
+                    "type": "Data",
+                    "format": "Parquet",
+                    "slot_size_array": slot_sizes,
+                    "source": DATA_DIR + "train/_file_list.txt",
+                    "eval_source": DATA_DIR + "valid/_file_list.txt",
+                    "check": "Sum",
+                    "label": {"top": "label", "label_dim": 1},
+                    "dense": {"top": "dense", "dense_dim": 0},
+                    "sparse": [
+                        {
+                            "top": "data1",
+                            "type": "DistributedSlot",
+                            "max_feature_num_per_sample": len(CATEGORICAL_COLUMNS),
+                            "slot_num": len(slot_sizes),
+                        }
+                    ],
+                },
+                {
+                    "name": "sparse_embedding1",
+                    "type": "DistributedSlotSparseEmbeddingHash",
+                    "bottom": "data1",
+                    "top": "sparse_embedding1",
+                    "sparse_embedding_hparam": {
+                        "max_vocabulary_size_per_gpu": total_cardinality,
+                        "embedding_vec_size": 16,
+                        "combiner": 0,
+                    },
+                },
+                {
+                    "name": "reshape1",
+                    "type": "Reshape",
+                    "bottom": "sparse_embedding1",
+                    "top": "reshape1",
+                    "leading_dim": 32,
```

Comment:
If you use 3 slots(slot_num=3) to train the model and the _embedding_vec_size_ is 16, then the **leading_dim** should be 3*16=48,

Reply:
Updated that. Now it works with 48 leading_dim.